src/rgw/rgw_sal.h

```diff
@@ -1365,6 +1370,39 @@ class Lifecycle {
 						       const std::string& oid,
 						       const std::string& cookie) = 0;
 };
+/**
+ * @brief Abstraction for a Notification configuration entry
+ *
+ * RGW can generate notifications for various events, and this represents a
+ * configuration entry for such a notification.  Config can exist globally or
+ * at the bucket level.
+ */
+class NotificationConfig {
+public:
+  class Result {
+  public:
+    virtual ~Result() = default;
+    virtual void encode(bufferlist& bl) const = 0;
+    virtual void decode(bufferlist::const_iterator& bl) = 0;
+    virtual void dump(Formatter *f) const = 0;
+  };
+  NotificationConfig() {}
+
+  virtual ~NotificationConfig() = default;
+
+  /** Read the config entry into @a data and (optionally) @a objv_tracker */
+  virtual int read(Result* data, RGWObjVersionTracker* objv_tracker) = 0;
```

Comment:
missing optional_yield and DoutPrefixProvider

Reply:
thanks for the comment, adding these found places where code was not yielding!